src/sparseml/openpifpaf/train.py

```diff
@@ -151,11 +157,37 @@ def main():
 
     datamodule = openpifpaf.datasets.factory(args.dataset)
 
+    """
+    Cases for checkpointing:
+    1. No checkpoint - no changes to base flow
+    2. Checkpoint from local file OR zoo stub:
+        a. set args.checkpoint to None to get a random network
+        b. manually load in state dict in _load_managers_from_checkpoint
+    """
+    if args.checkpoint:
+        if args.checkpoint.startswith("zoo:"):
+            args.checkpoint = download_framework_model_by_recipe_type(
+                Model(args.checkpoint)
+            )
+        checkpoint = torch.load(args.checkpoint, map_location="cpu")
+        # NOTE: set basenet instead of checkpoint so we get a randomized network
+        args.basenet = checkpoint["meta"]["args"]["basenet"]
```

Comment:
Is this always set by openpifpaf?

Reply:
It's from the Network cli. the basenet and checkpoint arguments are mutually exclusive, so if checkpoint is set there are checks that basenet is not set and vice versa.

The network factory code is essentially:
```
if args.basenet:
    return net, epoch
else:
    load_from_checkpoint
```

So the code here is basically making sure we never use their checkpoint loading logic, since we always want to be applying structure and loading weights ourselves